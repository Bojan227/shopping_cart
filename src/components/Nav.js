import React from 'react';
import {Link} from "react-router-dom"
import cart from '../images/cart.png'

export default function Nav(props){

    return (

        <nav>
            <ul>
                <Link to='/'>
                    <li>Home</li>
                </Link>

                <Link to='/shop'>
                    <li>Shop</li>
                </Link>

                <img src={cart} alt='cart' onClick={props.toggleCart} />
               
            </ul>

        </nav>


    )

}