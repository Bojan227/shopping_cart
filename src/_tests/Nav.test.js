import React from 'react';
import { screen, render } from "@testing-library/react";
import "@testing-library/jest-dom"; 
import { BrowserRouter } from 'react-router-dom';
import Nav from '../components/Nav';


describe('testing Nav bar', ()=>{
    it('if length of shopping by goes up by 1', ()=>{    
        render(
            <BrowserRouter>
                <Nav bagItems={['1', '2']}/>
            </BrowserRouter>
        )

        expect(screen.getByTestId('count-items').textContent).toMatch('2')
    })

    it('if Nav section renders correctly', ()=>{    
       render(
            <BrowserRouter>
                <Nav bagItems={['1', '2']}/>
            </BrowserRouter>
        )

        expect(<Nav />).toMatchSnapshot()
    })
    



})