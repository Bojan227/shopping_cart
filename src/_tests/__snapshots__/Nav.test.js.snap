// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing Nav bar if Nav section renders correctly 1`] = `<Nav />`;
